test/trento_web/controllers/v1/discovery_controller_test.exs

```diff
@@ -2,13 +2,15 @@ defmodule TrentoWeb.V1.DiscoveryControllerTest do
   use TrentoWeb.ConnCase, async: true
 
   describe "discovery" do
-    test "collect action should return bad request when the payload is invalid", %{conn: conn} do
+    test "collect action should return bad request when the event is unknown", %{conn: conn} do
```

Comment:
`collect action should return bad request...` is in conflict with what the api actually returns, a 422.